### PR TITLE
added deleteOne model to resourceContoller

### PR DESCRIPTION
**Pull request by cpmander**

### CODE REVIEW

File: server/controllers/resourceController.js [hunk 1]
```diff
@@ -45,24 +44,63 @@ resourceController.createResource = async (req, res, next) => {
...
         type: 'db insert error',
         err,
       })
     );
   }
+  return next();
+};
 
+resourceController.deleteResource = async (req, res, next) => {
+  const required = ['id', 'user', 'date'];
```
Comment (seadubs) [inline server/controllers/resourceController.js]: For delete, we should only need the unique id of the resource to delete, right?

File: server/controllers/resourceController.js [hunk 2]
```diff
@@ -45,24 +44,63 @@ resourceController.createResource = async (req, res, next) => {
...
+    return next(
+      createErr({
+        method: 'deleteResource',
+        type: 'data validation error',
+        err: 'request body contained invalid data',
+      })
+    );
+  }
+  try {
+    const dbRes = await Resource.deleteOne({ id: id, user: user, date: date, text: text, embeds: embeds, attachments: attachments, score: score, resources: resources, subject: subject, category: category });
```
Comment (seadubs) [inline server/controllers/resourceController.js]: Is the reason we're doing a query on all fields to make sure we're deleting the right record, or should the unique id be enough?

**Review by seadubs — APPROVED** [review]
lgtm

File: server/controllers/resourceController.js (hunk 1 above)
Comment (cpmander) [reply server/controllers/resourceController.js]: Resolved by #47

File: server/controllers/resourceController.js (hunk 2 above)
Comment (cpmander) [reply server/controllers/resourceController.js]: Resolved by #47

**Pull request merged**